[Bug] generate-swagger job from github actions is broken

Error is below. You can see an example in this [link](https://github.com/TBD54566975/ssi-service/actions/runs/4285933716/jobs/7464841987). This is preventing updates to our documentation. 

The stuff to highlight in the error is `ParseComment error in file /home/runner/work/ssi-service/ssi-service/pkg/server/router/webhook.go :cannot find type definition: GetSupportedNounsResponse`
```
go: downloading github.com/sirupsen/logrus v1.9.0
go: downloading golang.org/x/crypto v0.6.0
go: downloading golang.org/x/sys v0.5.0
go: downloading golang.org/x/term v0.5.0
Could not find swag on PATH or in GOPATH/bin
time="2023-03-01T[18](https://github.com/TBD54566975/ssi-service/actions/runs/4285933716/jobs/7464841987#step:6:19):29:23Z" level=warning msg="Error running command: /opt/hostedtoolcache/go/1.[20](https://github.com/TBD54566975/ssi-service/actions/runs/4285933716/jobs/7464841987#step:6:21).1/x64/bin/go get -u github.com/swaggo/swag/cmd/swag@v1.8.7" error="exit status 1"
time="20[23](https://github.com/TBD54566975/ssi-service/actions/runs/4285933716/jobs/7464841987#step:6:24)-03-01T18:29:39Z" level=info msg="Successfully installed github.com/swaggo/swag/cmd/swag@v1.8.7"
2023/03/01 18:29:59 ParseComment error in file /home/runner/work/ssi-service/ssi-service/pkg/server/router/webhook.go :cannot find type definition: GetSupportedNounsResponse
Error: running "swag init -g cmd/main.go --pd -o doc -ot yaml --parseDepth=3 --parseGoList=false" failed with exit code 1
Error: Process completed with exit code 1.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -49,6 +49,9 @@ jobs:
     - name: Test
       run: mage CITest
 
+    - name: Generate Swagger
+      run: mage spec
+
     - name: Upload coverage to Codecov
       uses: codecov/codecov-action@v3
 

--- a/doc/swagger.yaml
+++ b/doc/swagger.yaml
@@ -11,6 +11,8 @@ definitions:
         type: string
       revoked:
         type: boolean
+      suspended:
+        type: boolean
     type: object
   credential.CredentialSchema:
     properties:
@@ -224,6 +226,27 @@ definitions:
         type: boolean
     type: object
   did.Service:
+    properties:
+      accept:
+        items:
+          type: string
+        type: array
+      id:
+        type: string
+      routingKeys:
+        items:
+          type: string
+        type: array
+      serviceEndpoint:
+        description: |-
+          A string, map, or set composed of one or more strings and/or maps
+          All string values must be valid URIs
+      type:
+        type: string
+    required:
+    - id
+    - serviceEndpoint
+    - type
     type: object
   did.VerificationMethod:
     properties:
@@ -531,6 +554,11 @@ definitions:
         description: The subject id.
         example: did:key:z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp
         type: string
+      suspendable:
+        description: |-
+          Whether this credential can be suspended. When true, the created VC will have the "credentialStatus"
+          property set.
+        type: boolean
     required:
     - data
     - issuer
@@ -688,6 +716,32 @@ definitions:
     required:
     - submissionJwt
     type: object
+  github.com_tbd54566975_ssi-service_pkg_server_router.CreateWebhookRequest:
+    properties:
+      noun:
+        description: 'The noun (entity) for the new webhook.eg: Credential'
+        type: string
+      urls:
+        description: The URLS to post the output of this request to Noun.Verb action
+          to.
+        items:
+          type: string
+        type: array
+      verb:
+        description: 'The verb for the new webhook.eg: Create'
+        type: string
+    required:
+    - noun
+    - urls
+    - verb
+    type: object
+  github.com_tbd54566975_ssi-service_pkg_server_router.CreateWebhookResponse:
+    properties:
+      id:
+        type: string
+      webhook:
+        $ref: '#/definitions/webhook.Webhook'
+    type: object
   github.com_tbd54566975_ssi-service_pkg_server_router.GetApplicationResponse:
     properties:
       application:
@@ -726,6 +780,9 @@ definitions:
     properties:
       revoked:
         description: Whether the credential has been revoked.
+        type: boolean
+      suspended:
+        description: Whether the credential has been suspended.
         type: boolean
     type: object
   github.com_tbd54566975_ssi-service_pkg_server_router.GetCredentialsResponse:
@@ -864,6 +921,20 @@ definitions:
           along with the credentials presented.
     required:
     - status
+    type: object
+  github.com_tbd54566975_ssi-service_pkg_server_router.GetWebhookResponse:
+    properties:
+      id:
+        type: string
+      webhook:
+        $ref: '#/definitions/webhook.Webhook'
+    type: object
+  github.com_tbd54566975_ssi-service_pkg_server_router.GetWebhooksResponse:
+    properties:
+      webhooks:
+        items:
+          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetWebhookResponse'
+        type: array
     type: object
   github.com_tbd54566975_ssi-service_pkg_server_router.ListDefinitionsRequest:
     type: object
@@ -1006,13 +1077,15 @@ definitions:
           The new revoked status of this credential. The status will be saved in the encodedList of the StatusList2021
           credential associated with this VC.
         type: boolean
-    required:
-    - revoked
+      suspended:
+        type: boolean
     type: object
   github.com_tbd54566975_ssi-service_pkg_server_router.UpdateCredentialStatusResponse:
     properties:
       revoked:
         description: The updated status of this credential.
+        type: boolean
+      suspended:
         type: boolean
     type: object
   github.com_tbd54566975_ssi-service_pkg_server_router.VerifyCredentialRequest:
@@ -1292,6 +1365,11 @@ definitions:
         description: The subject id.
         example: did:key:z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp
         type: string
+      suspendable:
+        description: |-
+          Whether this credential can be suspended. When true, the created VC will have the "credentialStatus"
+          property set.
+        type: boolean
     required:
     - data
     - issuer
@@ -1450,6 +1528,32 @@ definitions:
     required:
     - submissionJwt
     type: object
+  pkg_server_router.CreateWebhookRequest:
+    properties:
+      noun:
+        description: 'The noun (entity) for the new webhook.eg: Credential'
+        type: string
+      urls:
+        description: The URLS to post the output of this request to Noun.Verb action
+          to.
+        items:
+          type: string
+        type: array
+      verb:
+        description: 'The verb for the new webhook.eg: Create'
+        type: string
+    required:
+    - noun
+    - urls
+    - verb
+    type: object
+  pkg_server_router.CreateWebhookResponse:
+    properties:
+      id:
+        type: string
+      webhook:
+        $ref: '#/definitions/webhook.Webhook'
+    type: object
   pkg_server_router.GetApplicationResponse:
     properties:
       application:
@@ -1489,6 +1593,9 @@ definitions:
     properties:
       revoked:
         description: Whether the credential has been revoked.
+        type: boolean
+      suspended:
+        description: Whether the credential has been suspended.
         type: boolean
     type: object
   pkg_server_router.GetCredentialsResponse:
@@ -1627,6 +1734,20 @@ definitions:
           along with the credentials presented.
     required:
     - status
+    type: object
+  pkg_server_router.GetWebhookResponse:
+    properties:
+      id:
+        type: string
+      webhook:
+        $ref: '#/definitions/webhook.Webhook'
+    type: object
+  pkg_server_router.GetWebhooksResponse:
+    properties:
+      webhooks:
+        items:
+          $ref: '#/definitions/pkg_server_router.GetWebhookResponse'
+        type: array
     type: object
   pkg_server_router.ListDefinitionsRequest:
     type: object
@@ -1769,13 +1890,15 @@ definitions:
           The new revoked status of this credential. The status will be saved in the encodedList of the StatusList2021
           credential associated with this VC.
         type: boolean
-    required:
-    - revoked
+      suspended:
+        type: boolean
     type: object
   pkg_server_router.UpdateCredentialStatusResponse:
     properties:
       revoked:
         description: The updated status of this credential.
+        type: boolean
+      suspended:
         type: boolean
     type: object
   pkg_server_router.VerifyCredentialRequest:
@@ -1925,6 +2048,42 @@ definitions:
       version:
         type: string
     type: object
+  webhook.GetSupportedNounsResponse:
+    properties:
+      nouns:
+        items:
+          description: 'The noun (entity) for the new webhook.eg: Credential'
+          type: string
+        type: array
+    type: object
+  webhook.GetSupportedVerbsResponse:
+    properties:
+      verbs:
+        items:
+          description: 'The verb for the new webhook.eg: Create'
+          type: string
+        type: array
+    type: object
+  webhook.Webhook:
+    properties:
+      id:
+        type: string
+      noun:
+        description: 'The noun (entity) for the new webhook.eg: Credential'
+        type: string
+      urls:
+        items:
+          type: string
+        type: array
+      verb:
+        description: 'The verb for the new webhook.eg: Create'
+        type: string
+    required:
+    - id
+    - noun
+    - urls
+    - verb
+    type: object
 host: localhost:8080
 info:
   contact:
@@ -1949,7 +2108,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetHealthCheckResponse'
+            $ref: '#/definitions/pkg_server_router.GetHealthCheckResponse'
       summary: Health Check
       tags:
       - HealthCheck
@@ -2227,7 +2386,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/pkg_server_router.GetDIDMethodsResponse'
+            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetDIDMethodsResponse'
       summary: Get DID Methods
       tags:
       - DecentralizedIdentityAPI
@@ -2248,7 +2407,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/pkg_server_router.GetDIDsByMethodResponse'
+            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetDIDsByMethodResponse'
         "400":
           description: Bad request
           schema:
@@ -2266,7 +2425,7 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/pkg_server_router.CreateDIDByMethodRequest'
+          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.CreateDIDByMethodRequest'
       - description: Method
         in: path
         name: method
@@ -2278,7 +2437,7 @@ paths:
         "201":
           description: Created
           schema:
-            $ref: '#/definitions/pkg_server_router.CreateDIDByMethodResponse'
+            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.CreateDIDByMethodResponse'
         "400":
           description: Bad request
           schema:
@@ -2301,7 +2460,7 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/pkg_server_router.CreateDIDByMethodRequest'
+          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.CreateDIDByMethodRequest'
       - description: Method
         in: path
         name: method
@@ -2318,7 +2477,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/pkg_server_router.GetDIDByMethodResponse'
+            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetDIDByMethodResponse'
         "400":
           description: Bad request
           schema:
@@ -2343,7 +2502,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/pkg_server_router.ResolveDIDResponse'
+            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.ResolveDIDResponse'
         "400":
           description: Bad request
           schema:
@@ -2445,7 +2604,7 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/pkg_server_router.StoreKeyRequest'
+          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.StoreKeyRequest'
       produces:
       - application/json
       responses:
@@ -2463,6 +2622,28 @@ paths:
       tags:
       - KeyStoreAPI
   /v1/keys/{id}:
+    delete:
+      consumes:
+      - application/json
+      description: Revoke a stored key
+      parameters:
+      - description: ID
+        in: path
+        name: id
+        required: true
+        type: string
+      produces:
+      - application/json
+      responses:
+        "200":
+          description: OK
+        "400":
+          description: Bad request
+          schema:
+            type: string
+      summary: Revoke Key
+      tags:
+      - KeyStoreAPI
     get:
       consumes:
       - application/json
@@ -2479,7 +2660,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/pkg_server_router.GetKeyDetailsResponse'
+            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetKeyDetailsResponse'
         "400":
           description: Bad request
           schema:
@@ -2512,7 +2693,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetManifestsResponse'
+            $ref: '#/definitions/pkg_server_router.GetManifestsResponse'
         "400":
           description: Bad request
           schema:
@@ -2534,14 +2715,14 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.CreateManifestRequest'
+          $ref: '#/definitions/pkg_server_router.CreateManifestRequest'
       produces:
       - application/json
       responses:
         "201":
           description: Created
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.CreateManifestResponse'
+            $ref: '#/definitions/pkg_server_router.CreateManifestResponse'
         "400":
           description: Bad request
           schema:
@@ -2598,7 +2779,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetManifestResponse'
+            $ref: '#/definitions/pkg_server_router.GetManifestResponse'
         "400":
           description: Bad request
           schema:
@@ -2610,19 +2791,14 @@ paths:
     get:
       consumes:
       - application/json
-      description: Checks for the presence of a query parameter and calls the associated
-        filtered get method
+      description: Gets all the existing applications.
       produces:
       - application/json
       responses:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetApplicationsResponse'
-        "400":
-          description: Bad request
-          schema:
-            type: string
+            $ref: '#/definitions/pkg_server_router.GetApplicationsResponse'
         "500":
           description: Internal server error
           schema:
@@ -2641,7 +2817,7 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.SubmitApplicationRequest'
+          $ref: '#/definitions/pkg_server_router.SubmitApplicationRequest'
       produces:
       - application/json
       responses:
@@ -2649,7 +2825,7 @@ paths:
           description: Operation with a SubmitApplicationResponse type in the `result.response`
             field.
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.Operation'
+            $ref: '#/definitions/pkg_server_router.Operation'
         "400":
           description: Bad request
           schema:
@@ -2706,7 +2882,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetApplicationResponse'
+            $ref: '#/definitions/pkg_server_router.GetApplicationResponse'
         "400":
           description: Bad request
           schema:
@@ -2725,14 +2901,14 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.ReviewApplicationRequest'
+          $ref: '#/definitions/pkg_server_router.ReviewApplicationRequest'
       produces:
       - application/json
       responses:
         "201":
           description: Credential Response
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.SubmitApplicationResponse'
+            $ref: '#/definitions/pkg_server_router.SubmitApplicationResponse'
         "400":
           description: Bad request
           schema:
@@ -2756,11 +2932,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetResponsesResponse'
-        "400":
-          description: Bad request
-          schema:
-            type: string
+            $ref: '#/definitions/pkg_server_router.GetResponsesResponse'
         "500":
           description: Internal server error
           schema:
@@ -2813,9 +2985,13 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetResponseResponse'
+            $ref: '#/definitions/pkg_server_router.GetResponseResponse'
         "400":
           description: Bad request
+          schema:
+            type: string
+        "500":
+          description: Internal server error
           schema:
             type: string
       summary: Get response
@@ -2832,14 +3008,14 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/pkg_server_router.GetOperationsRequest'
+          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetOperationsRequest'
       produces:
       - application/json
       responses:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/pkg_server_router.GetOperationsResponse'
+            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetOperationsResponse'
         "400":
           description: Bad request
           schema:
@@ -2868,7 +3044,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/pkg_server_router.Operation'
+            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.Operation'
         "400":
           description: Bad request
           schema:
@@ -2897,7 +3073,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/pkg_server_router.Operation'
+            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.Operation'
         "400":
           description: Bad request
           schema:
@@ -2920,14 +3096,14 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.CreatePresentationDefinitionRequest'
+          $ref: '#/definitions/pkg_server_router.CreatePresentationDefinitionRequest'
       produces:
       - application/json
       responses:
         "201":
           description: Created
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.CreatePresentationDefinitionResponse'
+            $ref: '#/definitions/pkg_server_router.CreatePresentationDefinitionResponse'
         "400":
           description: Bad request
           schema:
@@ -2984,7 +3160,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetPresentationDefinitionResponse'
+            $ref: '#/definitions/pkg_server_router.GetPresentationDefinitionResponse'
         "400":
           description: Bad request
           schema:
@@ -3003,14 +3179,14 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.ListDefinitionsRequest'
+          $ref: '#/definitions/pkg_server_router.ListDefinitionsRequest'
       produces:
       - application/json
       responses:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.ListDefinitionsResponse'
+            $ref: '#/definitions/pkg_server_router.ListDefinitionsResponse'
         "400":
           description: Bad request
           schema:
@@ -3034,14 +3210,14 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.ListSubmissionRequest'
+          $ref: '#/definitions/pkg_server_router.ListSubmissionRequest'
       produces:
       - application/json
       responses:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.ListSubmissionResponse'
+            $ref: '#/definitions/pkg_server_router.ListSubmissionResponse'
         "400":
           description: Bad request
           schema:
@@ -3063,14 +3239,14 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.CreateSubmissionRequest'
+          $ref: '#/definitions/pkg_server_router.CreateSubmissionRequest'
       produces:
       - application/json
       responses:
         "201":
           description: The type of response is Submission once the operation has finished.
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.Operation'
+            $ref: '#/definitions/pkg_server_router.Operation'
         "400":
           description: Bad request
           schema:
@@ -3099,7 +3275,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetSubmissionResponse'
+            $ref: '#/definitions/pkg_server_router.GetSubmissionResponse'
         "400":
           description: Bad request
           schema:
@@ -3120,14 +3296,14 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.ReviewSubmissionRequest'
+          $ref: '#/definitions/pkg_server_router.ReviewSubmissionRequest'
       produces:
       - application/json
       responses:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.ReviewSubmissionResponse'
+            $ref: '#/definitions/pkg_server_router.ReviewSubmissionResponse'
         "400":
           description: Bad request
           schema:
@@ -3150,7 +3326,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetSchemasResponse'
+            $ref: '#/definitions/pkg_server_router.GetSchemasResponse'
         "500":
           description: Internal server error
           schema:
@@ -3168,14 +3344,14 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.CreateSchemaRequest'
+          $ref: '#/definitions/pkg_server_router.CreateSchemaRequest'
       produces:
       - application/json
       responses:
         "201":
           description: Created
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.CreateSchemaResponse'
+            $ref: '#/definitions/pkg_server_router.CreateSchemaResponse'
         "400":
           description: Bad request
           schema:
@@ -3232,7 +3408,7 @@ paths:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.GetSchemaResponse'
+            $ref: '#/definitions/pkg_server_router.GetSchemaResponse'
         "400":
           description: Bad request
           schema:
@@ -3251,14 +3427,14 @@ paths:
         name: request
         required: true
         schema:
-          $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.VerifySchemaRequest'
+          $ref: '#/definitions/pkg_server_router.VerifySchemaRequest'
       produces:
       - application/json
       responses:
         "200":
           description: OK
           schema:
-            $ref: '#/definitions/github.com_tbd54566975_ssi-service_pkg_server_router.VerifySchemaResponse'
+            $ref: '#/definitions/pkg_server_router.VerifySchemaResponse'
         "400":
           description: Bad request
           schema:
@@ -3266,4 +3442,135 @@ paths:
       summary: Verify Schema
       tags:
       - SchemaAPI
+  /v1/webhooks:
+    get:
+      consumes:
+      - application/json
+      description: Get webhooks
+      produces:
+      - application/json
+      responses:
+        "200":
+          description: OK
+          schema:
+            $ref: '#/definitions/pkg_server_router.GetWebhooksResponse'
+        "500":
+          description: Internal server error
+          schema:
+            type: string
+      summary: Get Webhooks
+      tags:
+      - WebhookAPI
+    put:
+      consumes:
+      - application/json
+      description: Create webhook
+      parameters:
+      - description: request body
+        in: body
+        name: request
+        required: true
+        schema:
+          $ref: '#/definitions/pkg_server_router.CreateWebhookRequest'
+      produces:
+      - application/json
+      responses:
+        "201":
+          description: Created
+          schema:
+            $ref: '#/definitions/pkg_server_router.CreateWebhookResponse'
+        "400":
+          description: Bad request
+          schema:
+            type: string
+        "500":
+          description: Internal server error
+          schema:
+            type: string
+      summary: Create Webhook
+      tags:
+      - WebhookAPI
+  /v1/webhooks/{id}:
+    delete:
+      consumes:
+      - application/json
+      description: Delete a webhook by its ID
+      parameters:
+      - description: ID
+        in: path
+        name: id
+        required: true
+        type: string
+      produces:
+      - application/json
+      responses:
+        "200":
+          description: OK
+          schema:
+            type: string
+        "400":
+          description: Bad request
+          schema:
+            type: string
+        "500":
+          description: Internal server error
+          schema:
+            type: string
+      summary: Delete Webhook
+      tags:
+      - WebhookAPI
+    get:
+      consumes:
+      - application/json
+      description: Get a webhook by its ID
+      parameters:
+      - description: ID
+        in: path
+        name: id
+        required: true
+        type: string
+      produces:
+      - application/json
+      responses:
+        "200":
+          description: OK
+          schema:
+            $ref: '#/definitions/pkg_server_router.GetWebhookResponse'
+        "400":
+          description: Bad request
+          schema:
+            type: string
+      summary: Get Webhook
+      tags:
+      - WebhookAPI
+  /v1/webhooks/nouns:
+    get:
+      consumes:
+      - application/json
+      description: Get supported nouns for webhook generation
+      produces:
+      - application/json
+      responses:
+        "200":
+          description: OK
+          schema:
+            $ref: '#/definitions/webhook.GetSupportedNounsResponse'
+      summary: Get Supported Nouns
+      tags:
+      - WebhookAPI
+  /v1/webhooks/verbs:
+    get:
+      consumes:
+      - application/json
+      description: Get supported verbs for webhook generation
+      produces:
+      - application/json
+      responses:
+        "200":
+          description: OK
+          schema:
+            $ref: '#/definitions/webhook.GetSupportedVerbsResponse'
+      summary: Get Supported Verbs
+      tags:
+      - WebhookAPI
 swagger: "2.0"

--- a/pkg/server/router/keystore.go
+++ b/pkg/server/router/keystore.go
@@ -139,7 +139,7 @@ func (ksr *KeyStoreRouter) GetKeyDetails(ctx context.Context, w http.ResponseWri
 // @Tags        KeyStoreAPI
 // @Accept      json
 // @Produce     json
-// @Param       id  path     string true "ID"
+// @Param       id path string true "ID"
 // @Success     200
 // @Failure     400 {string} string "Bad request"
 // @Router      /v1/keys/{id} [delete]

--- a/pkg/server/router/webhook.go
+++ b/pkg/server/router/webhook.go
@@ -98,7 +98,7 @@ type GetWebhookResponse struct {
 // @Success     200 {object} GetWebhookResponse
 // @Failure     400 {string} string "Bad request"
 // @Router      /v1/webhooks/{id} [get]
-func (wr WebhookRouter) GetWebhook(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+func (wr WebhookRouter) GetWebhook(ctx context.Context, w http.ResponseWriter, _ *http.Request) error {
 	id := framework.GetParam(ctx, IDParam)
 	if id == nil {
 		errMsg := "cannot get webhook without ID parameter"
@@ -131,7 +131,7 @@ type GetWebhooksResponse struct {
 // @Success     200 {object} GetWebhooksResponse
 // @Failure     500 {string} string "Internal server error"
 // @Router      /v1/webhooks [get]
-func (wr WebhookRouter) GetWebhooks(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+func (wr WebhookRouter) GetWebhooks(ctx context.Context, w http.ResponseWriter, _ *http.Request) error {
 	gotWebhooks, err := wr.service.GetWebhooks(ctx)
 	if err != nil {
 		errMsg := "could not get webhooks"
@@ -184,10 +184,9 @@ func (wr WebhookRouter) DeleteWebhook(ctx context.Context, w http.ResponseWriter
 // @Tags        WebhookAPI
 // @Accept      json
 // @Produce     json
-// @Success     200 {object} GetSupportedNounsResponse
-// @Failure     500 {string} string "Internal server error"
+// @Success     200 {object} webhook.GetSupportedNounsResponse
 // @Router      /v1/webhooks/nouns [get]
-func (wr WebhookRouter) GetSupportedNouns(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+func (wr WebhookRouter) GetSupportedNouns(ctx context.Context, w http.ResponseWriter, _ *http.Request) error {
 	nouns := wr.service.GetSupportedNouns()
 	return framework.Respond(ctx, w, nouns, http.StatusOK)
 }
@@ -199,10 +198,9 @@ func (wr WebhookRouter) GetSupportedNouns(ctx context.Context, w http.ResponseWr
 // @Tags        WebhookAPI
 // @Accept      json
 // @Produce     json
-// @Success     200 {object} GetSupportedVerbsResponse
-// @Failure     500 {string} string "Internal server error"
+// @Success     200 {object} webhook.GetSupportedVerbsResponse
 // @Router      /v1/webhooks/verbs [get]
-func (wr WebhookRouter) GetSupportedVerbs(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+func (wr WebhookRouter) GetSupportedVerbs(ctx context.Context, w http.ResponseWriter, _ *http.Request) error {
 	verbs := wr.service.GetSupportedVerbs()
 	return framework.Respond(ctx, w, verbs, http.StatusOK)
 }

--- a/pkg/service/webhook/model.go
+++ b/pkg/service/webhook/model.go
@@ -1,20 +1,24 @@
 package webhook
 
-// In the context of webhooks, it's common to use noun.verb notation to describe events,
+// Noun In the context of webhooks, it's common to use noun.verb notation to describe events,
 // such as "credential.create" or "schema.delete".
 type Noun string
 type Verb string
 
 // Supported Nouns
-const Credential = Noun("Credential")
-const DID = Noun("DID")
-const Manifest = Noun("Manifest")
-const Schema = Noun("Schema")
-const Presentation = Noun("Presentation")
+const (
+	Credential   = Noun("Credential")
+	DID          = Noun("DID")
+	Manifest     = Noun("Manifest")
+	Schema       = Noun("Schema")
+	Presentation = Noun("Presentation")
+)
 
 // Supported Verbs
-const Create = Verb("Create")
-const Delete = Verb("Delete")
+const (
+	Create = Verb("Create")
+	Delete = Verb("Delete")
+)
 
 type Webhook struct {
 	ID   string   `json:"id" validate:"required"`

